core/src/main/java/org/frankframework/pipes/XmlSwitch.java

```diff
@@ -119,81 +116,70 @@ public void stop() {
 	 */
 	@Override
 	public PipeRunResult doPipe(Message message, PipeLineSession session) throws PipeRunException {
-		String forward="";
+		String forward = "";
 		PipeForward pipeForward;
-		if(StringUtils.isNotEmpty(getForwardNameSessionKey())) {
+
+		if (StringUtils.isNotEmpty(getForwardNameSessionKey())) {
 			forward = session.getString(getForwardNameSessionKey());
-		} else if(!(StringUtils.isEmpty(getXpathExpression()) && StringUtils.isEmpty(getStyleSheetName())) || StringUtils.isEmpty(getSessionKey())) {
+		} else if (StringUtils.isNotEmpty(getXpathExpression()) || StringUtils.isNotEmpty(getStyleSheetName()) || StringUtils.isEmpty(getGetInputFromSessionKey())) {
 			try {
-				Map<String,Object> parametervalues = null;
+				Map<String, Object> parametervalues = null;
 				ParameterList parameterList = getParameterList();
+
 				if (!parameterList.isEmpty()) {
 					parametervalues = parameterList.getValues(message, session, isNamespaceAware()).getValueMap();
 				}
-				if(StringUtils.isNotEmpty(getSessionKey())) {
-					forward = transformerPool.transform(session.getMessage(getSessionKey()), parametervalues, isNamespaceAware());
-				} else {
-					message.preserve();
-					forward = transformerPool.transform(message, parametervalues, isNamespaceAware());
-				}
+
+				message.preserve();
+				forward = transformerPool.transform(message, parametervalues, isNamespaceAware());
 			} catch (Throwable e) {
 				throw new PipeRunException(this, "got exception on transformation", e);
 			}
-		} else if(StringUtils.isNotEmpty(getSessionKey())) {
-			forward = session.getString(getSessionKey());
+		} else {
+			try {
+				// Use the message as forward if none of the cases above apply
+				forward = message.asString();
```

Comment:
In de oude code kom je volgens mij op regel 137 uit, en moet je `forward = transformerPool.transform(message, parametervalues, isNamespaceAware());` doen. Nu pak je alleen de input, klopt dit wel?